hpy/debug/src/debug_internal.h

```diff
@@ -36,6 +35,21 @@
 
    Each DebugHandle has a "generation", which is just a int to be able to get
    only the handles which were created after a certain point.
+
+   DHPys/DebugHandles are memory-managed by using a free list:
+
+     - info->open_handles is a list of all DHPys which are currently open
+
+     - DHPy_close() moves a DHPy from info->open_handles to info->closed_handles
+
+     - if closed_handles is too big, the oldest DHPy is freed by DHPy_free()
+
+     - to allocate memory for a new DHPy, DHPy_open() does the following:
+
+         * if closed_handles is empty, it malloc()s memory for a new DHPy
+
+         * if it's not empty, it reuses the memory of the oldest DHPy in the
```

Comment:
This description makes it sound like `DHPy_open` will always use a closed handle even if there is only one handle in `info->closed_handles`? I sort of expected `DHPy_open` to always malloc a new handle and then for the handle to be freed when `info->closed_handles` eventually gets too big.

Reply:
you are correct,t the comment was wrong. I tried to fix it in 2b08586